Sources/VLCSettingsController.m

```diff
@@ -199,9 +201,43 @@ - (UIView *)settingsViewController:(id<IASKViewController>)settingsViewControlle
     return header;
 }
 
+- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
+{
+    [tableView deselectRowAtIndexPath:indexPath animated:YES];
+    IASKSpecifier *specifier = [self.settingsReader specifierForIndexPath:indexPath];
+
+    if ([specifier.type isEqualToString: kIASKPSMultiValueSpecifier]) {
+        [self displayActionSheetFor:specifier];
+    } else {
+        [super tableView:tableView didSelectRowAtIndexPath:indexPath];
+    }
+}
+
 - (NSString *)tableView:(UITableView *)tableView titleForHeaderInSection:(NSInteger)section
 {
     return nil;
 }
 
+- (void)displayActionSheetFor:(IASKSpecifier *)specifier
+{
+    VLCActionSheet *actionSheet = [[VLCActionSheet alloc] init];
+    actionSheet.modalPresentationStyle = UIModalPresentationCustom;
+    [actionSheet.collectionView registerClass:[VLCSettingsSheetCell class] forCellWithReuseIdentifier:VLCSettingsSheetCell.identifier];
+    
+    specifierManager = [[VLCSettingsSpecifierManager alloc] initWithSpecifier:specifier settingsReader:self.settingsReader settingsStore:self.settingsStore];
+    actionSheet.delegate = specifierManager;
+    actionSheet.dataSource = specifierManager;
+    
+    [actionSheet setActionWithClosure:^(id item) {
+        [self.settingsStore setObject:item forKey:specifier.key];
+        [self.settingsStore synchronize];
+        NSDictionary *info = [NSDictionary dictionaryWithObject:item forKey:[specifier key]];
+        [[NSNotificationCenter defaultCenter] postNotificationName:kIASKAppSettingChanged
+                                                            object:[specifier key]
+                                                          userInfo:info];
```

Comment:
do you need to fire the notification here? Is it not done in the setter of the settingsstore?

Reply:
Wow, I didn't know that :D